### ISSUE TITLE
Intermittent segfault with list_string_agg on dictionary-compressed column

### ISSUE THREAD

**Issue by paulewog** [duckdb/duckdb#5781]
### What happens?

A segmentation fault occurs when using list_string_agg on a dictionary-compressed VARCHAR[] column. It doesn't happen every time, but it seems to happen more often than not.

### To Reproduce

Using the attached test.db:

```
duckdb test.db 'SELECT list_string_agg(ips) FROM Hosts'
```

Note that I created the table using the attached CSV of IPs like this:
```
duckdb test.db "pragma force_compression='dictionary'; CREATE TABLE Hosts (ips varchar[]); INSERT INTO Hosts SELECT * FROM 'ips.csv';"
```

[ips.csv.gz](https://github.com/duckdb/duckdb/files/10297715/ips.csv.gz)
[test.db.gz](https://github.com/duckdb/duckdb/files/10297716/test.db.gz)

### OS:

OSX x64, Linux x64

### DuckDB Version:

0.6.1

### DuckDB Client:

CLI

### Full Name:

Paul Ellsworth

### Affiliation:

Tenable, Inc.

### Have you tried this on the latest `master` branch?

- [X] I agree

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

**Comment by paulewog** [comment]
I should also mention that building the same table, except using _uncompressed_ instead of _dictionary_ works fine. :)

**Comment by Mytherin** [comment]
Thanks for the report! I have pushed a fix in #5854
